pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/CreateSegmentCommand.java

```diff
@@ -413,6 +445,32 @@ public void run() {
     return executor.awaitTermination(1, TimeUnit.HOURS);
   }
 
+  private boolean verifySegment(String indexDirStr) {
+    SegmentMetadataImpl segmentMetadata;
+    File localTempDir;
+    try {
+      localTempDir = new File("segmentVerification", org.apache.pinot.common.utils.FileUtils.getRandomFileName());
+      FileSystem.get(URI.create(indexDirStr), new Configuration())
+          .copyToLocalFile(new Path(indexDirStr), new Path(localTempDir.toString()));
+    } catch (IOException e) {
+      LOGGER.error("Failed to copy segment {} to local directory for verification.", indexDirStr, e);
+      return false;
+    }
+    try {
```

Comment:
I don't think this will work. You need to untar the segment first.
I would recommend untar the segment, then use `ImmutableSegmentLoader.load()` to verify if you can load the segment.

Reply:
`CreateSegment` script will not tar files.  `UploadSegment` script will do so. See: https://github.com/apache/incubator-pinot/blob/master/pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/UploadSegmentCommand.java#L128